crates/luminol-audio/src/wrapper.rs

```diff
@@ -82,117 +80,102 @@ impl AudioWrapper {
     pub fn play(
         &self,
         path: impl AsRef<camino::Utf8Path>,
+        filesystem: &impl luminol_filesystem::FileSystem,
         volume: u8,
         pitch: u8,
         source: Source,
-    ) -> Result<(), String> {
+    ) -> Result<()> {
         // We have to load the file on the current thread,
         // otherwise if we read the file in the main thread of a web browser
         // we will block the main thread
         let path = path.as_ref();
-        let mut file = state!()
-            .filesystem
-            .open_file(path, filesystem::OpenFlags::Read)
-            .map_err(|e| e.to_string())?;
-        let length = state!()
-            .filesystem
-            .metadata(path)
-            .map_err(|e| e.to_string())?
-            .size as usize;
-        let mut vec = vec![0; length];
-        file.read(&mut vec[..]).map_err(|e| e.to_string())?;
+        let file = filesystem.open_file(path, luminol_filesystem::OpenFlags::Read)?;
```

Comment:
Changing this part of the code like this to pass the file handle directly to Audio running on the main thread is going to cause a crash because the read method on this file handle is blocking. I can't think of a better way around this than reading the entire file beforehand.

Reply:
Oh that's annoying. I think I can fix this pretty easily by relaxing `Audio` to only take something implementing `Read` + `Seek` rather than `File` and then just passing a cursor